kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/store/DbWireRecordStore.java

```diff
@@ -395,124 +414,120 @@ private void reconcileColumns(final String tableName, final WireRecord wireRecor
-            this.tickHandle = this.executorService.schedule(new Runnable() {
+    private PreparedStatement prepareStatement(Connection connection, String tableName,
+            final Map<String, TypedValue<?>> properties, long timestamp) throws SQLException {
 
-                /** {@inheritDoc} */
-                @Override
-                public void run() {
-                    DbWireRecordStore.this.clear(noOfRecordsToKeep);
-                }
-            }, cleanUpRate, TimeUnit.SECONDS);
+        final String sqlTableName = this.dbHelper.sanitizeSqlTableAndColumnName(tableName);
+        final StringBuilder sbCols = new StringBuilder();
+        final StringBuilder sbVals = new StringBuilder();
+
+        // add the timestamp
+        sbCols.append("TIMESTAMP");
+        sbVals.append("?");
+
+        int i = 2;
+        for (Entry<String, TypedValue<?>> entry : properties.entrySet()) {
+            final String sqlColName = this.dbHelper.sanitizeSqlTableAndColumnName(entry.getKey());
+            sbCols.append(", " + sqlColName);
+            sbVals.append(", ?");
         }
-    }
 
-    /**
-     * Stores the provided wire record in the database
-     *
-     * @param wireRecord
-     *            the wire record to be stored
-     * @throws NullPointerException
-     *             if the provided argument is null
-     */
-    private void store(final WireRecord wireRecord) {
-        requireNonNull(wireRecord, s_message.wireRecordNonNull());
-        boolean inserted = false;
-        int retryCount = 0;
-        final String tableName = this.options.getTableName();
-        do {
-            try {
-                this.insertDataRecord(tableName, wireRecord);
-                inserted = true;
-            } catch (final SQLException e) {
-                s_logger.error(s_message.insertionFailed() + ThrowableUtil.stackTraceAsString(e));
-                try {
-                    if ((tableName != null) && (!tableName.isEmpty())) {
-                        retryCount++;
-                        this.reconcileTable(tableName);
-                        this.reconcileColumns(tableName, wireRecord);
-                    }
-                } catch (final SQLException ee) {
-                    s_logger.error(s_message.errorStoring() + ee);
-                }
+        logger.debug(message.storingRecord(sqlTableName));
+        final String sqlInsert = MessageFormat.format(SQL_INSERT_RECORD, sqlTableName, sbCols.toString(),
+                sbVals.toString());
+        final PreparedStatement stmt = connection.prepareStatement(sqlInsert);
+        stmt.setLong(1, timestamp);
+
+        for (Entry<String, TypedValue<?>> entry : properties.entrySet()) {
+            final DataType dataType = entry.getValue().getType();
+            final Object value = entry.getValue();
+            switch (dataType) {
+            case BOOLEAN:
+                logger.debug(message.storeBoolean(((BooleanValue) value).getValue()));
```

Comment:
remove all the debug statements as these are more verbose and hence removed from DB Filter as well